internal/server/v1/dlq/service.go

```diff
@@ -24,3 +28,16 @@ func NewService(client entropyv1beta1rpc.ResourceServiceClient, gcsClient gcs.Bl
 		cfg:       cfg,
 	}
 }
+
+func (s *Service) getDlqJob(ctx context.Context, jobURN string) (*models.DlqJob, error) {
+	res, err := s.client.GetResource(ctx, &entropyv1beta1.GetResourceRequest{Urn: jobURN})
+	if err != nil {
+		return nil, err
```

Comment:
let's create a new error called, `ErrJobNotFound` similar with https://github.com/goto/dex/blob/main/internal/server/v1/optimus/errors.go#L6.

And return the error **only if** the error from the client is not found.


And in the handler, we can return 404 if service returns that new error. example: https://github.com/goto/dex/blob/main/internal/server/v1/optimus/handler.go#L26

Reply:
Entropy should send 404 if the job is not found right? We are getting 500 right now from Entropy.

Reply:
tried via postman and entropy returned 404. Since entropy is using grpc-gateway, if we are hitting via grpc, it would be translated to `NOT_FOUND` with `code = 5`.

we can test it like this
https://github.com/goto/dex/blob/main/internal/server/v1/firehose/firehose.go#L84

Reply:
yeah tested it out, although status was 500 still the check for NotFound from status worked.